lib/iris/coords.py

```diff
@@ -1948,24 +1948,35 @@ def collapsed(self, dims_to_collapse=None):
         if np.issubdtype(self.dtype, np.str_):
             # Collapse the coordinate by serializing the points and
             # bounds as strings.
-            def serialize(x):
-                return "|".join([str(i) for i in x.flatten()])
+            def serialize(x, axis):
+                if axis is None:
+                    return "|".join(str(i) for i in x.flatten())
+                # np.apply_along_axis does not work with str.join, so we
+                # need to loop through the array directly. First move (possibly
+                # multiple) axis of interest to trailing dim(s), then make a 2D
+                # array we can loop through.
+                work_array = np.moveaxis(x, axis, range(-len(axis), 0))
+                out_shape = work_array.shape[: -len(axis)]
+                work_array = work_array.reshape(np.prod(out_shape), -1)
+
+                joined = []
+                for arr_slice in work_array:
+                    joined.append(serialize(arr_slice, None))
+
+                return np.array(joined).reshape(out_shape)
 
             bounds = None
             if self.has_bounds():
-                shape = self._bounds_dm.shape[1:]
                 bounds = []
-                for index in np.ndindex(shape):
-                    index_slice = (slice(None),) + tuple(index)
-                    bounds.append(serialize(self.bounds[index_slice]))
```

Comment:
If I've understood the old slicing method, it and the new method are consistent for the basic case of 2D bounds (n_points, n_bounds).  For higher dimensions, the old method doesn't look right to me: if we had a 2D (n, m) coordinated with bounds (n, m, n_bounds), it looks like we would get `m * n_bounds` elements in the list.  But I **think** we want just `n_bounds` elements.

I didn't even know bounded string coords were a thing!